R/predictions.R

```diff
@@ -48,10 +48,11 @@ prediction <- function(dt, prediction_zero, explainer) {
   stopifnot(nrow(explainer$x_test) == dt[, max(id)])
 
   # Predictions
-  dt[, p_hat := predict_model(explainer$model, newdata = .SD), .SDcols = cnms]
-  dt[id_combination == 1, p_hat := prediction_zero]
+  dt[!(n_features %in% c(0, ncol(explainer$x_test))),
+     p_hat := predict_model(explainer$model, newdata = .SD), .SDcols = cnms]
+  dt[n_features == 0, p_hat := prediction_zero]
   p_all <- predict_model(explainer$model, newdata = explainer$x_test)
-  dt[id_combination == max(id_combination), p_hat := p_all[id]]
```

Comment:
Why isn't this ok? Shouldn't `max(id_combination)` always be equal to the case with all variables?

Reply:
Could you give me an example so I could reproduce the problem?

Reply:
You are right. Not relevant anymore.

Reply:
I feel like these changes in `prediction` are unnecessary.  Here it is only 2 * number of test samples additional rows we are predicting on when doing it the old way. Searching through the entire table, using a non-key column, for those rows seems inefficient to me. The reason why we decided on using `id_combination` was that that is the column that is used throughout everything.

Reply:
Yes, sure, after the changes in prepare_data, there are far fewer extra rows this is done for. I now went back to `id_combination`, but rather extracts `p_all` from `dt`, as due to the changes in `prepare_data`, x_test will always be placed in those columns and not something randomly sampled. What do you think? If you are happy, we could also remove `n_features `from the export of prepare_data as it is not needed in `prediction.`

Reply:
@nikolase90 What do you think about the last changes here? If you think this approach is fine, I will modify `prepare_data` not to include `n_features`.